lua/autorun/server/sv_cfc_contact_forms.lua

```diff
@@ -32,6 +33,79 @@ local function groomSubmissionCounts()
 end
 timer.Create( 'CFC_GroomFormSubmissions', SUBMISSION_GROOM_INTERVAL, 0, groomSubmissionCounts )
 
+local function getPlayerCounts()
+    local playerCounts = {}
+
+    for _, ent in pairs( ents.GetAll() ) do
+        if IsValid( ent ) then
+            local className = ent:GetClass()
+            local entOwner = ent:CPPIGetOwner() or ent:GetOwner()
+            local ownerId = ( entOwner:IsPlayer() and entOwner:SteamID64() ) or 'world'
+
+            playerCounts[ownerId] = playerCounts[ownerId] or {}
+            playerCounts[ownerId][className] = ( playerCounts[ownerId][className] or 0 ) + 1
+        end
+    end
+
+    return playerCounts
+end
+
+local function getE2Information()
+    local playerE2Info = {}
+
+    for _, expression2 in pairs( ents.FindByClass( 'gmod_wire_expression2' ) ) do
+        if IsValid( expression2 ) then
+            local e2Owner = expression2:CPPIGetOwner()
+            local ownerId = ( e2Owner:IsPlayer() and e2Owner:SteamID64() ) or 'world'
```

Comment:
This may error if e2Owner isnt valid or is null, might be a good call to add `e2Owner and e2Owner:IsValid() and ...`